src/fs/fs.ts

```diff
@@ -320,6 +322,39 @@ export class FileSystem extends EventEmitter implements FlashDataSource {
     this.fileVersions.set(filename, current === undefined ? 1 : current + 1);
   }
 
+  async getPythonProject(): Promise<PythonProject> {
+    const projectName = await this.storage.projectName();
+    const project: PythonProject = {
+      files: {},
+      projectName,
+    };
+    for (const file of await this.storage.ls()) {
+      const data = await this.storage.read(file);
+      const content = new TextDecoder().decode(data);
+      project.files[file] = content;
```

Comment:
I think we don't want to be doing this if we want binary file support.
Rather than taking the bytes and converting to a string we want to go bytes -> base64 (and in reverse when reading from a Python project).

Reply:
File content encoded to base64 - [`2a6ae34`](https://github.com/microbit-foundation/python-editor-next/pull/626/commits/2a6ae34c0d7cd2b57103434db58ee04b5c850639)